### ISSUE TITLE
[stripe] When i try to make a payment with stripe and wrong card number i got this error

### ISSUE THREAD

**Issue by adrian410** [Payum/Payum#601]
Hi,

When i try to make a payment with stripe and wrong card number i got this error :

```
"The either card token or customer id has to be set."
n vendor\payum\payum\src\Payum\Stripe\Action\Api\CreateChargeAction.php at line 46
```

Exemple of wrong card number : `4000000000000127`

I got this error when i use your template "Action/obtainèjs_token.html.twig".
Only with it, because "obtain_checkout_token.html.twig" catch the error with ajax with the Stripe Pop-Up.

```
public function viewAction(Request $request)
    {
        echo "ok";
        exit();
        $token = $this->getPayum()->getHttpRequestVerifier()->verify($request);

        $gateway = $this->getPayum()->getGateway($token->getGatewayName());

        try {
            $gateway->execute(new Sync($token));
        } catch (RequestNotSupportedException $e) {}



        $gateway->execute($status = new GetHumanStatus($token));


        $payment = $status->getFirstModel();


        $em = $this->container->get('doctrine')->getEntityManager(); 
        $className = $em->getClassMetadata(get_class($payment))->getName();
        $details=$payment->getDetails();


        try{
            $value=$status->getValue();
        }catch(\Exception $error){
            $value=null;
        }

        try{
            $gatewayName=$token->getGatewayName();
        }catch(\Exception $error){
            $gatewayName=null;
        }

        $refundToken = null;
        if ($status->isCaptured() || $status->isAuthorized()) {
            $refundToken = $this->getPayum()->getTokenFactory()->createRefundToken(
                $token->getGatewayName(),
                $status->getFirstModel(),
                $request->getUri()
            );
        }



        $captureToken = null;
        if ($status->isAuthorized() && ($value!=null && refundToken!=null)) {
            $captureToken = $this->getPayum()->getTokenFactory()->createCaptureToken(
                $token->getGatewayName(),
                $status->getFirstModel(),
                $request->getUri()
            );
        }

        
        $details = $status->getFirstModel();
        if ($details instanceof  DetailsAggregateInterface) {
            $details = $details->getDetails();
        }

        if ($details instanceof  \Traversable) {
            $details = iterator_to_array($details);
        }

        if($gatewayName!=null && $value!=null){
            // Sauvegarde de la carte de crédit
            $user= $this->get('security.token_storage')->getToken()->getUser();
            $gateway->execute($getToken = new GetCreditCardToken($payment));
            $this->get('credit_card.service')->save($getToken,$user);
        }




        return $this->render('CroaPaymentBundle:Details:view.html.twig', array(
            'status' => $value,
            'payment' => htmlspecialchars(json_encode($details, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE)),
            'gatewayTitle' => ucwords(str_replace(array('_', '-'), ' ', $gatewayName)),
            'refundToken' => $refundToken,
            'captureToken' => $captureToken,
        ));
    }
```

I put a test with an exit at the begining of the controller, and can't even get the < echo "ok" >, it show me the error directely.

Also thanks for the update with the credit card save with Stripe, it's working well :)

Hope you will figure it our where it come from.

Best regards,
Adrian Abramczyk

**Comment by frechina** [comment]
I have same issue

**Comment by makasim** [comment]
great, @frechina would like to volunteer and fix the bug?

**Comment by frechina** [comment]
sure! I will in few minutes

**Comment by frechina** [comment]
Fixed in #642
